src/main/host/descriptor/socket/inet/mod.rs

```diff
@@ -221,6 +221,27 @@ impl PartialEq for InetSocket {
 
 impl Eq for InetSocket {}
 
+impl std::hash::Hash for InetSocket {
+    /// Returns a hash for the socket based on its address, and not the socket's state. Two
+    /// different sockets with the same state will return different hashes, and the same socket will
+    /// return the same hash even after being mutated.
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        // To match the `Eq` behaviour of `InetSocket`, the hashes of two sockets *must* be equal if
+        // the types are the same and the Arc's pointers are equal, and the hashes *should not* be
+        // equal if the two types are not the same or the Arc's pointers are not equal. We do return
+        // the same hash if the two types are different but the pointers are equal, but this is
+        // allowed by the `Hash` trait (it's just a hash collision) and we should never run into
+        // this without a variant containing a ZST, which wouldn't make sense in the context of
```

Comment:
Dumb question: what is "ZST"?

Reply:
Changed "ZST" to "zero-sized type". What this is trying to say is that usually two different objects will have different memory addresses. For example with `let x = Box::new(0)` and `let y = Box::new(0)`, the two integers will have different memory allocations and different addresses. But this isn't true in the case of zero-sized types. For example:

```rust
let x = Box::new(());
let y = Box::new([0u8; 0]);
println!("{:p}, {:p}", &*x as *const _, &*y as *const _);
```

prints "0x1, 0x1", which would indicate that the data stored by each `Box` have the same address even though they aren't the same objects or the same type.